verticles/graphql/src/main/java/com/gentics/mesh/graphql/dataloader/NodeDataLoader.java

```diff
@@ -376,13 +373,13 @@ public boolean equals(Object o) {
 					&& pagingId == context.pagingId
 					&& Objects.equals(languageTags, context.languageTags)
 					&& Objects.equals(
-							maybeNativeFilter.map(Objects::toString).orElse(StringUtils.EMPTY), 
-							context.maybeNativeFilter.map(Objects::toString).orElse(StringUtils.EMPTY));
+							maybeNativeFilter.map(FilterOperation::toSql).orElse(StringUtils.EMPTY), 
```

Comment:
Long story short: when a `NodeDataLoader` fills up the fetch context, it forms a `HashMap`, where the key is formed from this `Context`. The `FilterOperation` is however misbehaving, as must be compared by its SQL representation, and not default `toString()`, so the similar operations will be merged into one context, which reduces the run time and query count.

Reply:
The UT for it is placed in an Enterprise, since it is the one who suffered.